### PR TITLE
Send emails

### PR DESCRIPTION
**Pull request by fkurusu**

### CODE REVIEW

File: app/controllers/password_resets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).first
```
Comment (brandoncc) [inline app/controllers/password_resets_controller.rb]: Another way to this is:

```ruby
user = User.find_by(token: params[:id])
```

File: app/models/user.rb [hunk 2]
```diff
@@ -10,6 +10,8 @@ class User < ActiveRecord::Base
   validates :password, presence: true, length: { minimum: 8}, on: :create
   validates :full_name, presence: true
 
+  # before_create :generate_token
```
Comment (brandoncc) [inline app/models/user.rb]: Commented code is almost always unneeded code. Make sure you clean up code you don't need anymore.

File: app/views/forgot_passwords/new.html.haml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+%section.forgot_password.container
+  .row
+    .span10.offset1
```
Comment (brandoncc) [inline app/views/forgot_passwords/new.html.haml]: `span*` tags are not valid in Bootstrap 3 which this project uses.

File: db/migrate/20150930213653_generate_migration_for_existing_users.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class GenerateMigrationForExistingUsers < ActiveRecord::Migration
+  def change
+    User.all.each do |user|
```
Comment (brandoncc) [inline db/migrate/20150930213653_generate_migration_for_existing_users.rb]: When you are iterating through `all` records, you should use `find_each`. That will run through in batches (of 1000 by default), hopefully preventing your system from running out of memory.

File: spec/controllers/forgot_passwords_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
+require 'spec_helper'
+
+describe ForgotPasswordsController do
+  describe "POST create" do
+    context "with blank input" do
+      let(:post_action) { post :create, email_address: '' }
```
Comment (brandoncc) [inline spec/controllers/forgot_passwords_controller_spec.rb]: You can move this to a `before` block and then you don't have to manually call it in your specs. I do kind of like how explicit your current way is though, while DRYing up the code.....very nice :smile:

File: spec/controllers/password_resets_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      end
+
+      it "updates the user's password" do
+        post :create, password: "new_password", token: user.token
+        expect(user.reload.authenticate('new_password')).to be_truthy
+      end
+
+      it "sets the flash success message" do
+        post :create, password: "new_password", token: user.token
+        expect(flash[:success]).to eq "Your password has been changed. Please sign in."
```
Comment (brandoncc) [inline spec/controllers/password_resets_controller_spec.rb]: Be careful tightly coupling your implementation to your specs like this. I would recommend just making sure that the flash key is present, not necessarily verifying what it contains.

**Comment by fkurusu** [comment]
It was applied some suggestions that will be merged to master branch.

**Pull request merged**